backend/zerowaste/src/main/java/com/zerowaste/models/product/Product.java

```diff
@@ -187,6 +187,13 @@ public void setDeletedAt(LocalDate deletedAt) {
         this.deletedAt = deletedAt;
     }
 
+    public void setPromotions(Set<Promotion> promotions2) {
+        if (promotions2 == null) {
+            throw new IllegalArgumentException("Promoções não podem ser nulas.");
+        }
+        this.promotions = Collections.unmodifiableSet(promotions2);
+    }
+
```

Comment:
Vendo esse trecho, acabo de perceber que no meu PR (#71) eu não coloquei o getter/setter de 
`Set<Promotion> promotions` em `Product`. Boa.

Mas, com isso, os métodos `getPromotions` e `setPromotions` devem ser os habituais de getters and setters mesmo.

Já a validação dos dados deve ser feita nos Services.

```suggestion
    public Set<Promotion> getPromotions() {
        return promotions;
    }

    public void setPromotions(Set<Promotion> promotions) {
        this.promotions = promotions;
    }

```